### DISCUSSION TITLE
is there any way to add module for ingress without compiling the image ?

### DISCUSSION THREAD

**Discussion by rthamrin** [Kong/kubernetes-ingress-controller#2479] [Q&A] [answered]
I have a question to ask about adding a new module to the ingress, [compression module](https://github.com/tokers/zstd-nginx-module).  However it is written in C. so I have to convert the file to module library (so)

#gcc -c fPIC ngx_http_zstd_static_module.c -o ngx_http_zstd_static_module.so

```
ngx_http_zstd_static_module.c:7:10: fatal error: ngx_config.h: No such file or directory

    7 | #include <ngx_config.h>

      |          ^~~~~~~~~~~~~~

compilation terminated.
```
is anyone knows how to add the module to ingress ?

thank you

**Comment by shaneutt** [comment, marked as answer]
Here check out this older forum post this would seem to cover the exact question you have here: https://discuss.konghq.com/t/using-nginx-modules-in-kong/1688

Let us know if that's helpful, or if you have further questions or concerns?
